software/speller/data_collection_platform/dcp/bci/stream.py

```diff
@@ -1 +1,123 @@
-# TODO stephen
+"""
+This file will allow the application to connect to the BCI process
+and collect the brain signals being streamed
+This code is based on
+1. https://github.com/OpenBCI/OpenBCI_GUI/blob/master/Networking-Test-Kit/LSL/lslStreamTest.py
+2. the previous years stream.py found at NeuroTechX-McGill-2021/software/data_collection_platform/backend/dcp/bci/stream.py
+"""
+
+from pylsl import StreamInlet, resolve_stream
+import time 
+import logging
+import os
+
+log_path = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.realpath(__file__)))), "logs",
+                        "bci.log")
+
+logging.basicConfig(filename=log_path,
+                    level=logging.INFO,
+                    format="%(asctime)s %(name)s %(levelname)s %(message)s")
+
+logger = logging.getLogger(__name__)
+logger.info('test logger')
+
+duration = 5
```

Comment:
I'd move this `duration` variable as a parameter of the `testLSLSamplingRate` method (can default at 5)

Reply:
Good point, will do